src/interceptors/ClientRequest/NodeClientRequest.ts

```diff
@@ -146,15 +169,7 @@ export class NodeClientRequest extends ClientRequest {
      *
      * For that reason, we have to keep an internal flag for a finished request.
      */
-    this.isRequestSent = true
-
-    const requestId = uuidv4()
-
-    const [chunk, encoding, callback] = normalizeClientRequestEndArgs(...args)
-    this.logger.info('normalized arguments:', { chunk, encoding, callback })
-
-    // Write the last request body chunk passed to the "end()" method.
-    this.writeRequestBodyChunk(chunk, encoding || undefined)
+    this.state = HttpClientInternalState.Sent
```

Comment:
I'm replacing `isRequestSent` boolean flag with the `Sent` state of the request.